### ISSUE TITLE
[customer]nodeset create dracut.* file in sles statelite cost too much time

### ISSUE THREAD

**Issue by bybai** [xcat2/xcat-core#3224]
This description is from Galid Berman lenovo mail:
because with 24 service nodes, we get 50 (!!) such files for each node (2 per service node and 2 per xcat master), the nodeset process is SO SLOW… removing the files will only makes it worse, because it will put extra load on that directory.

Under my investigate:
`genimage` and `nodeset` can create `rootimg/tmp/dracut.* ` files, `liteimg` will copy and pack the image, in `liteimg` enhancement, `liteimg` is to check and delete `rootimg/tmp/dracut.*` files first. Since genimage is used for both diskless and diskful, I do not delete these files in `nodeset` or `genimage`, enhance `liteimg` is enough. see issue #3142 

We use other code to replace dracut command in nodeset to check if dracut is used or not. So that there is no dracut.* files are created by nodeset, and the same time nodeset does not need to call shell command and system command, it may improve the nodeset performance.

We can update the function in code /opt/xcat/lib/perl/xCAT_plugin/sles.pm,  use this issue to trace enhancement;
```
sub using_dracut
{
    my $rootimgdir = shift;
    #my $chkcmd = "chroot $rootimgdir/rootimg dracut --list-modules";
    #my $rc = system($chkcmd);
    #if ($rc) {
    if ( -f "$rootimgdir/rootimg/etc/dracut.conf" ) {
        return 1;
    } else {
        return 0;
    }
}
```

**Comment by bybai** [comment]
Hi @daniceexi , we use `chroot $rootimgdir/rootimg dracut --list-modules` to check if `dracut` is used or not in `nodeset` command. `nodeset` call this command `dracut --list-modules` for sles statelite, this command will generate 1 dracut.* tmp files under rootimg/tmp files for one node, `nodeset` call this command twice, it will create 2 dracut.* files when executing `nodeset` for one node , these 2 files are 0 bytes and useless.  In Galid env, he has 24 service nodes, he gets 50 (!!) such dracut.* files for each node (2 per service node and 2 per xcat master), he has total 3.5K compute nodes, `nodeset` generate more or less 50 dracut.* files for each node so he get lots of dracut.* files,  the nodeset process is SO SLOW…   In my enhancement, I used `rootimg/etc/dracut.conf` to replace `dracut --list-modules` command to check if there is dracut or not, if there is dracut.conf, there is `dracut` and we will use dracut, and tmp dracut.* files will not be created, and `nodeset` will deduce call `system` command to execute `dracut` command twice, it can improve the performance. I looked into code for redhat, the function is totally different with sles, so this problem only existed in sles statelite `nodeset`. In my first pull request, I mis-return 0 and 1, based on my unit test, I commit the second code changes. That should work well. Thanks.
